### ISSUE TITLE
SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversableParameterTypeHintSpecification fails when it references an interface implementation

### ISSUE THREAD

**Issue by dhaarbrink** [slevomat/coding-standard#175]
Given the following interface:

```php
interface NormalizerInterface
{
    public function normalize($object, $format = null, array $context = array());
    public function supportsNormalization($data, $format = null);
}
```
This interface is defined by symfony and hence can't be changed.

When I implement this interface I can't specify typehints in the declarations. 

I think this sniff should check if the method is part of an interface and if it is, ignore it. When the sniff encounters the interface itself it should complain there.

**Comment by ondrejmirtes** [comment]
Unfortunately, this is not possible in PHP_CodeSniffer because CodeSniffer works in a single file scope and does not have access to reflection. See https://medium.com/@ondrejmirtes/three-pillars-of-static-analysis-in-php-f3f5d7bfd61b for more details.

Slevomat CS solves this by offering suppressing rules locally, [see README](https://github.com/slevomat/coding-standard#suppressing-sniffs-locally) with example for your exact use case.

**Comment by dhaarbrink** [comment]
That's too bad.
Thanks for pointing me in the right direction!
